planner/core/issuetest/planner_issue_test.go

```diff
@@ -106,3 +106,16 @@ func TestIssue44051(t *testing.T) {
 	rs := tk.MustQuery("WITH tmp AS (SELECT t2.* FROM t2) SELECT * FROM t1 WHERE t1.id = (select id from tmp where id = 1) or t1.id = (select id from tmp where id = 2) or t1.id = (select id from tmp where id = 3)")
 	rs.Sort().Check(testkit.Rows("1 <nil> <nil> <nil>", "2 <nil> <nil> <nil>", "3 <nil> <nil> <nil>"))
 }
+
+func TestIssue45007(t *testing.T) {
+	store := testkit.CreateMockStore(t)
```

Comment:
Maybe add `failpoint.Enable("github.com/pingcap/tidb/planner/core/forceDynamicPrune", `return(true)`)` here, I found it not always use dynamic pruning for this situation.